sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3038,14 +3046,20 @@ def _execute_task_n_nodes(self, handle: ResourceHandle, task: task_lib.Task,
         else:
             num_actual_nodes = task.num_nodes
 
+        cluster_external_ips = backend_utils.get_node_ips(handle.cluster_yaml,
+                                                          handle.launched_nodes,
+                                                          handle=handle)
         cluster_internal_ips = backend_utils.get_node_ips(handle.cluster_yaml,
                                                           handle.launched_nodes,
                                                           handle=handle,
                                                           get_internal_ips=True)
-        # Ensure head node is the first element, then sort the remaining IPs
-        # for stableness
-        cluster_ips_sorted = [cluster_internal_ips[0]] + sorted(
-            cluster_internal_ips[1:])
+        cluster_ip_pairs = list(zip(cluster_internal_ips, cluster_external_ips))
+
+        # Ensure head node is the first element, then sort based on the external
+        # IPs for stableness
+        cluster_ips_pairs_sorted = [cluster_ip_pairs[0]] + sorted(
+            cluster_ip_pairs[1:], key=lambda x: x[1])
+        cluster_internal_ips_sorted = [x[0] for x in cluster_ips_pairs_sorted]
```

Comment:
nit: This name could suggest this list itself is sorted. How about `stable_cluster_internal_ips`? Same for the arg on L3072.

Reply:
agreed!